core/services/relay/evm/mercury/transmitter.go

```diff
@@ -31,6 +33,7 @@ import (
 
 const (
 	MaxTransmitQueueSize = 10_000
+	PruneFrequency       = 10 * time.Second
```

Comment:
I would suggest using `Prune` as a fallback insurance and deleting items simultaneously with queue eviction instead (line 86 of queue.go). This may help to reduce table contention. A bit more complex but the ideal solution would be something like an "eviction queue". You store up items to evict then purge them on a timer, every 15 seconds for example. If there are no items to evict, it does nothing.

If you do it this way, you can keep the full prune, but running it e.g. once per hour would be sufficient.